library/psa_crypto_driver_wrappers.c

```diff
@@ -381,27 +381,30 @@ psa_status_t psa_driver_wrapper_verify_hash(
 }
 
 /** Get the key buffer size required to store the key material of a key
- *  associated with an opaque driver without storage.
+ *  associated with an opaque driver.
  *
  * \param[in] attributes  The key attributes.
+ * \param[in] key_bits    The actual key bits used to deduce the size, given
```

Comment:
But is this not what we're already doing here ?, calculating based on data length instead of the attribute bits.
Additionally though not a data import the sizing is also required for derived keys if the key is meant to be stored as wrapped.
I would categorize the sizing as not restricted to  psa_import_key, but rather when we invoke psa_driver_wrapper_import_key. Since the sizing varies based on the driver implementation.

Reply:
> But is this not what we're already doing here ?, calculating based on data length instead of the attribute bits.
When importing a key, if the key size is not provided in the key attributes, my understanding is that you need the key data (not only the key data length) to determine properly the key buffer size. Determining the key buffer size may involve to parse the key import data.

> Additionally though not a data import the sizing is also required for derived keys if the key is meant to be stored as wrapped.
Yes but in that case you have the key size and type and the current signature of `psa_driver_wrapper_get_key_buffer_size()` is suitable, right ?

Reply:
I agree that the data should be passed in the case of an import.
Here the sizing from the old routine should suffice for opaque keys with a pre determined size.